frontend/views/containers/payments/PaymentsList.vue

```diff
@@ -143,6 +146,74 @@ export default {
     },
     symbol () {
       return currencies[this.groupSettings.mincomeCurrency].symbol
+    },
+    paymentsReceived () {
+      return 'TODO'
+    },
+    paymentsTodo () {
+      const monthlyPayments = this.groupMonthlyPayments
+      const cMonthstamp = currentMonthstamp()
+      const pMonthstamp = prevMonthstamp(cMonthstamp)
+      const paymentsTodo = []
+      const pastMonth = monthlyPayments[pMonthstamp]
+      // const currentDistribution = this.groupIncomeAdjustedDistribution
+      const currentDistribution = this.paymentsDistribution
+      const pledgeAmount = this.ourGroupProfile['pledgeAmount']
+      // first we need to check if there are any late payments that need to be added to the paymentsTodo list
+      if (pastMonth) {
+        const previousDistribution = pastMonth.lastAdjustedDistribution
+        // this creates a date at the end of the previous month
```

Comment:
A UI component (Vue, React, etc...) main responsibility is to map data to the UI, avoiding creating any extra complex logic, leaving that part to the "store / backend part", in our case the Vuex getters.

In my opinion, this logic could be a getter `this.paymentsLateDistribution` and then the UI only concacts both:

```js
return this.paymentsLateDistribution.concat(this.paymentsDistribution)
```

Another advantage of this approach is we can create a unit test to make sure this works as expected (since with Cypress it will be _a big challenge_ to simulate late payments.)

Reply:
Maybe. It makes sense to make it a getter only if it's used more than once. I'll think about this more when I get to writing tests. 👍